Utils/Dataflow/060_upload2virtuoso/uploadTTL.sh

```diff
@@ -158,14 +170,15 @@ upload_stream () {
   esac
 
   while true; do
-    while read -r -d "$delimiter" line; do
+    while read -r -d "$delimiter" line || [[ -n "$line" ]]; do
       n=`ps ax | grep 'curl' | grep "$HOST:$PORT" | grep -v 'grep' | wc -l`
       while [ $n -gt $CURL_N_MAX ]; do
         sleep $SLEEP
         n=`ps axf | grep 'curl' | grep "$HOST:$PORT" | grep -v 'grep' | wc -l`
       done
       echo "$line" | $cmd &>/dev/null || { echo "(ERROR) An error occured while uploading stream data." >&2; continue; } &
-      echo -n $'\6'
+      echo -ne EOMessage
+      echo -ne EOProcess
```

Comment:
And, now it seems to me that it is a bad idea to send evaluation of `$cmd` to the background.
See what can happen:
```
$ sleep 3 && command_that_fails || echo "Failed!" & \
> echo "End of process!"                                                                                       
[1] 16485
(INFO) End of process!
mgolosova@#artois-kiae:~/work/svn/git.dkb-project/Utils/Dataflow/069_upload2es$ command_that_fails: command not found
(ERROR) Failed!
```

We see the message saying that the process is finished (though it still goes on), and when after 3 seconds it fails, we see another message -- saying that the process has failed.

Right, we need to send EOP even when the process failed; but should we send it earlier than the process is finished?
It is a question of parallelization... if we do not send EOP before process is finished, we get no new data to run next process. But if we do, we can miss some records. I need to think about this a little...

For now it is OK to leave things as is; but it should be fixed somehow when we add manual position commit to the supervisor.